python_modules/dagster/dagster/_core/definitions/assets_job.py

```diff
@@ -231,12 +231,18 @@ def build_node_deps(
 def _has_cycles(deps: Dict[Union[str, NodeInvocation], Dict[str, IDependencyDefinition]]) -> bool:
     """Detect if there are cycles in a dependency dictionary."""
     try:
-        node_deps: Dict[Union[str, NodeInvocation], Set[str]] = {}
+        node_deps: Dict[str, Set[str]] = {}
```

Comment:
this change is unrelated -- it fixes a bug that I found when creating the example for the test.

Basically you would get the wrong error type when checking for cycles sometimes because toposort doesn't play nicely with keys that can be multiple different types.